### ISSUE TITLE
[FEATURE]: Access to purchasing shipping

### ISSUE THREAD

**Issue by chrisrogala** [etsy/open-api#645]
### Is there an existing ticket for this?

- [X] I have searched the existing tickets

### Is your feature request related to a problem? Please describe

I'm always frustrated when purchasing shipping through Etsy seller portal. It would be great if there was an API that all developers to purchase the shipping labels through the backend services instead of manually purchasing them in the portal. The reason for this is because of how manual the process is with purchasing, scheduling a pickup or creating a USPS scan sheet.

### Describe the solution you'd like

There is already an API available with the website. The endpoint is:

> https://www.etsy.com/api/v3/ajax/bespoke/shop/{{shopid}}/shipping-label/purchase-shipments

The body is:

> {
> 	"shipments": [
> 		{
> 			"notification_message": "{{Seller notification message}}",
> 			"order_id": "{{order number}}",
> 			"ship_to": {
> 				"city": "",
> 				"country_id": 209,
> 				"first_line": "",
> 				"name": "",
> 				"phone": "",
> 				"second_line": "",
> 				"state": "",
> 				"verification_state": 2,
> 				"zip": "",
> 				"is_confirmed": true
> 			},
> 			"ship_from": {
> 				"city": "",
> 				"country_id": 209,
> 				"first_line": "",
> 				"name": "",
> 				"phone": "",
> 				"second_line": "",
> 				"state": "",
> 				"verification_state": 0,
> 				"zip": ""
> 			},
> 			"package": {
> 				"mail_class": "first_class",
> 				"package_type": "parcel",
> 				"weight_units": "oz",
> 				"dimension_units": "in",
> 				"customs_items": [],
> 				"mailing_date": 0,
> 				"signature_confirmation": false,
> 				"label_type": 0,
> 				"insurance_value": 0,
> 				"insurance_provider_id": 3,
> 				"height": "1",
> 				"length": "11",
> 				"width": "9",
> 				"weight": 5
> 			},
> 			"uuid": "be044476-0df9-49fb-9655-e706d0ec5895",
> 			"origin_zip": "",
> 			"carrier_id": 1,
> 			"sort_order": 1
> 		}
> 	]
> }

There is another API for creating a scan sheet:

> https://www.etsy.com/api/v3/ajax/shop/{{shopId}}/shipping-labels/scan-form

The body is actually not JSON and instead it is form data.

There is another API that provides the ability to schedule a pickup:

> https://www.etsy.com/api/v3/ajax/shop/{{shopId}}/shipping-labels/usps-pickup

The body is:

{
	"address": {
		"name": "",
		"first_line": "",
		"second_line": "",
		"city": "",
		"state": "",
		"country_id": 209,
		"country_name": "",
		"zip": "",
		"phone": ""
	},
	"requested_pickup_date": 1667059200,
	"package_location": "Front Door",
	"special_instructions": "In box labeled USPS First Class",
	"excluded_shipping_label_ids": [
		
	]
}

### Describe any alternatives you've considered

Purchasing the shipping labels outside of Etsy.

### Additional context

While the portal works well for smaller shops, it is not designed for larger shops. This causes delays in our processing and results in our shop having to extend turn around time and turn customers away. Over the past two years, our shop stops accepting orders for Christmas deliveries because we do not have enough time to process more orders. If Etsy opened this up, it would improve our efficiency this holiday season so we could continue to take orders later into the year.

**Comment by etsyachristensen** [comment]
@chrisrogala Are you currently hitting those ajax endpoints programmatically? Or just seeing them in console?

**Comment by chrisrogala** [comment]
No, I'm watching the traffic. I'm fairly certain it is against Etsy's TOS to programmatically hit ajax endpoints. It would be huge if we could. I did a time study last year, and we spent at least 15-30 minutes each day searching for order numbers, purchasing shipping and creating scan sheets. Without the scan sheets, the USPS carrier will only scan 5 packages. Because of this, the tracking will not update until it arrives at the destination facility. We had hundreds of orders last year where the customer contacted us and we have to take time out of the day to research them and respond. In the end, we lost at least 16-32 man hours which means we had to turn sales away.

If you all are able to setup a proxy endpoint to authenticate us and then call the internal endpoint, that would be huge and would allow us to keep a shorter turnaround time since we can allocate those hours to create more product for Etsy buyers.

**Comment by etsyachristensen** [comment]
Sorry we could not accommodate this request this year. I doubt we can provide any type of proxy to those internal endpoints and yes they are (and should be always) locked down to our internal system.

I'll move this request to our discussion board in the Ideas category for easier tracking. Hopefully this will be something we can provide before next holiday season.
